source/PlayerInfoPanel.cpp

```diff
@@ -558,44 +671,60 @@ void PlayerInfoPanel::DrawFleet(const Rectangle &bounds)
 	Color dim = *GameData::Colors().Get("medium");
 	Color bright = *GameData::Colors().Get("bright");
 	Color elsewhere = *GameData::Colors().Get("dim");
-	Color dead(.4f, 0.f, 0.f, 0.f);
-	Color disabled(.5f, .3f, .1f, 0.f);
+	Color dead = *GameData::Colors().Get("dead");
+	Color special = *GameData::Colors().Get("special");
+	Color disabled = *GameData::Colors().Get("disabled");
 	
 	// Table attributes.
 	Table table;
-	table.AddColumn(0, Table::LEFT);
-	table.AddColumn(220, Table::LEFT);
-	table.AddColumn(350, Table::LEFT);
-	table.AddColumn(550, Table::RIGHT);
-	table.AddColumn(610, Table::RIGHT);
-	table.AddColumn(670, Table::RIGHT);
-	table.AddColumn(730, Table::RIGHT);
+	for(const auto &col: columns)
+		table.AddColumn(col.offset, col.align);
+	
 	table.SetUnderline(0, 730);
 	table.DrawAt(bounds.TopLeft() + Point(10., 8.));
+	table.DrawUnderline(dim);
 	
 	// Header row.
-	table.DrawUnderline(dim);
-	table.SetColor(bright);
-	table.Draw("ship");
-	table.Draw("model");
-	table.Draw("system");
-	table.Draw("shields");
-	table.Draw("hull");
-	table.Draw("fuel");
-	table.Draw("crew");
+	for(auto column = columns.begin(); column < columns.end(); ++column)
+	{
+		const auto currentColumn = *column;
```

Comment:
`auto` does not deduce reference types, so without `&` we would make a copy:
```suggestion
		const auto &currentColumn = *column;
```

(it would be better if we can avoid needing to use an explicit iterator at all, but I accept that we may need this style in this loop.)